Student/Student/Submissions/SubmissionDetails/SubmissionDetailsPresenter.swift

```diff
@@ -64,22 +70,54 @@ class SubmissionDetailsPresenter {
 
     func viewIsReady() {
         submissions.refresh(force: true)
-        assignment.refresh(force: true)
-        course.refresh(force: false)
+        assignment.refresh()
+        course.refresh()
+        view?.reloadNavBar()
     }
 
-    func submissionFor(attempt: Int) -> Submission? {
-        guard submissions.count > 0 else {
-            return nil
+    func update() {
+        let assignment = self.assignment.first
+        let submission = submissions.filter({ $0.attempt == selectedAttempt }).first ?? submissions.first
+        selectedAttempt = submission?.attempt ?? selectedAttempt
+        if submission?.attachments?.contains(where: { $0.id == selectedFileID }) != true {
+            selectedFileID = submission?.attachments?.sorted(by: { $0.id < $1.id }).first?.id
+        }
+
+        let assignmentChanged = (assignment != currentAssignment)
+        let fileIDChanged = (currentFileID != selectedFileID)
+        let submissionChanged = (submission != currentSubmission)
```

Comment:
Nitpick: Parens on these 3 lines are not needed. I do appreciate them being split up.

Reply:
yeah I debated that, but used them to make it obvious the `!=` has higher precedence than `=`.